client.go

```diff
@@ -21,14 +21,19 @@ type Client struct {
 	hostname   string
 	port       int
 	httpClient *http.Client
+	namespace  string
+	password   string
+	cachedAuth string
 }
 
 // NewClient returns a client ready for use
-func NewClient(hostname string, port int) *Client {
+func NewClient(hostname string, port int, namespace, password string) *Client {
 	return &Client{
 		hostname:   hostname,
 		port:       port,
 		httpClient: &http.Client{},
+		namespace:  namespace,
+		password:   password,
```

Comment:
Sorry to be annoying, but based on our conversation this afternoon the word "password" is going to change to "token" to encourage long random strings. That hasn't been merged yet though.

Reply:
will do.

of course, the name token will collide with the JWT token.

Reply:
I am in fact wondering if I should rename them all again just for fun...